dijet/tasks/plot_base.py

```diff
@@ -0,0 +1,99 @@
+from columnflow.tasks.framework.mixins import (
+    CalibratorsMixin, SelectorMixin, ProducersMixin,
+    VariablesMixin, DatasetsProcessesMixin, CategoriesMixin,
+)
+from columnflow.config_util import get_datasets_from_process
+from columnflow.util import dev_sandbox
+from dijet.tasks.base import DiJetTask
+
+
+class PlottingBaseTask(
+    DiJetTask,
+    DatasetsProcessesMixin,
+    CategoriesMixin,
+    VariablesMixin,
+    ProducersMixin,
+    SelectorMixin,
+    CalibratorsMixin,
+    ShiftTask,
+    law.LocalWorkflow,
+):
+    """
+    Base task to plot histogram from each step of the JER SF workflow.
+    An example implementation of how to handle the inputs in a run method can be
+    found in columnflow/tasks/histograms.py
+    """
+    sandbox = dev_sandbox(law.config.get("analysis", "default_columnar_sandbox"))
+
+    # Add nested sibling directories to output path
+    output_collection_cls = law.NestedSiblingFileCollection
+
+    # Category ID for methods
+    LOOKUP_CATEGORY_ID = {"sm": 1, "fe": 2}
+
+    def get_datasets(self) -> tuple[list[str], bool]:
+        """
+        Select datasets belonging to the `process` of the current branch task.
+        Returns a list of the dataset with requested Runs for data.
+        """
+
+        dataset_insts_from_process_data = get_datasets_from_process(
+            self.config_inst,
+            "data",
+            only_first=False,
+        )
+
+        # filter to contain only user-supplied datasets from data
+        datasets_from_process_data = [d.name for d in dataset_insts_from_process_data]
+        datasets_data_filtered = set(self.datasets).intersection(datasets_from_process_data)
+        # check that at least one user-supplied dataset matched
+        if not datasets_data_filtered:
+            raise RuntimeError(
+                "no single user-supplied dataset for data matched "
+                f"process `{dataset_insts_from_process_data}`",
+            )
+
+        # return filtered datasets
+        return list(datasets_data_filtered)
+
+    def extract_sample(self):
+        datasets = self.get_datasets()
+        runs = [dataset.replace("data_jetht_", "").upper() for dataset in datasets]
+        sample = "Run" + ("".join(sorted(runs)))
+        return sample
+
+    def store_parts(self):
+        parts = super().store_parts()
+        return parts
+
+    def eta_bin(self, eta_lo, eta_hi):
+        return fr"{eta_lo}<$\eta$<{eta_hi}"
+
+    def pt_bin(self, pt_lo, pt_hi):
+        spt = r"$p_\mathrm{T}$"
+        return fr"{pt_lo}<{spt}<{pt_hi}"
+
+    def alpha_bin(self, alpha):
+        salpha = r"$\alpha$<"
+        return fr"{salpha}{alpha}"
+
+    def dot_to_p(self, var):
+        return f"{var}".replace(".", "p")
+
+    def add_text(self, ax, x, y, text, offset=0, ho="left", ve="top"):
+        ax.text(
+            x, y - offset, text,
+            transform=ax.transAxes,
+            horizontalalignment=ho,
+            verticalalignment=ve,
+        )
```

Comment:
Suggest to use the more common variants `ha` and `va` for the alignment parameters, as also done in `matplotlib`.
```suggestion
    def add_text(self, ax, x, y, text, offset=0, ha="left", va="top"):
        ax.text(
            x, y - offset, text,
            transform=ax.transAxes,
            horizontalalignment=ha,
            verticalalignment=va,
        )
```

Reply:
added with the suggestion to move to `dijet/plotting`